modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala

```diff
@@ -90,10 +90,16 @@ object ExtractedDirectives {
       val plainCommentDirectives   = byKind(UsingDirectiveKind.PlainComment)
 
       val directivesPositionsOpt =
-        if (
-          codeDirectives.getFlattenedMap.isEmpty &&
-          specialCommentDirectives.getFlattenedMap.isEmpty &&
-          plainCommentDirectives.getFlattenedMap.isEmpty
+        if ( // check if there's no directives or all of them are target directives
+          codeDirectives.getFlattenedMap.keySet().asScala.forall(
+            _.toString.startsWith("target.")
+          ) &&
+          specialCommentDirectives.getFlattenedMap.keySet().asScala.forall(
+            _.toString.startsWith("target.")
+          ) &&
+          plainCommentDirectives.getFlattenedMap.keySet().asScala.forall(
+            _.toString.startsWith("target.")
```

Comment:
NIT extracting this to a util would greatly improve the readability of this `if` clause IMO
so basically, we could have a util object somewhere with:
```scala
extension (ud: UsingDirective) {
  def keySet: Set[String] = ud.getFlattenedMap.keySet().asScala
  def containsTargetDirectivesOnly: Boolean = ud.keySet.forall(_.toString.startsWith("target."))
}
```

and then we could

```suggestion
          codeDirectives.containsTargetDirectivesOnly &&
          specialCommentDirectives.containsTargetDirectivesOnly &&
          plainCommentDirectives.containsTargetDirectivesOnly
```